src/modules/cache.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2016 The Noise-rs Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use num_traits::Float;
+use math::{Point2, Point3, Point4};
+use NoiseModule;
+use std::cell::RefCell;
+
+/// Noise module that caches the last output value generated by the source
+/// module.
+///
+/// If an application passes input coordinates to the `get()` method that
+/// differs from the previously passed-in value, this noise module calls
+/// the source module to calculate the output value. The output value, as well
+/// as the input coordinates, are stored (cached) in this noise module.
+///
+/// If the application passes input coordinates to the `get()` method that is
+/// equal to the previously passed-in input coordinates, this noise module
+/// returns the cached output value without having the source module
+/// recalculate the output value.
+///
+/// Caching a noise module is useful if it is used as a source module for
+/// multiple noise modules. If a source module is not cached, the source
+/// module will redundantly calculate the same output value once for each
+/// noise module in which it is included.
+#[derive(Clone, Debug)]
+pub struct Cache<Source, T> {
+    /// Outputs the value to be cached.
+    pub source: Source,
+
+    value: RefCell<Option<T>>,
+
+    point: RefCell<Vec<T>>,
```

Comment:
No, because doing so would require specifying which point type it is and would lock the Cache module into supporting that specific point type on implementation, instead of keeping it generic over the set of potential point types, allowing the user to feed any point type they want into it. This would have knock on effects of locking the entire module chain into that specific point type if they have a Cache module as part of their chain.